### DISCUSSION TITLE
High CPU usage

### DISCUSSION THREAD

**Discussion by lupin3rd** [brunoherbelin/vimix#124] [Q&A]
Seem that there is an high CPU usage also with all videos deactivated... Why?
How can i understand what video impact on cpu usage?
i attach an image of a session that i load with all video stopped... i have 3 vimix process that use a lot of CPU
![Screenshot_2024-02-18_23-41-46](https://github.com/brunoherbelin/vimix/assets/1455316/37918606-6bed-47ea-a25d-ebb21541079d)

**Comment by lupin3rd** [comment]
@brunoherbelin in your experience, what is the best video codec to use with Vimix (in terms of performance) with minor CPU impact?

**Reply by MoorPoor** [reply]
I'm using apple prores if possible, but just because it has alpha channel. I can load a lot of those and CPU keeps cool.
just one CPU Core is usually gooing up.  

May I ask witch system you are using? If Linux Xorg or Wayland?
I can not use Vimix in Wayland without nuclear station next to me ;)

**Reply by lupin3rd** [reply]
i'm using Ubuntu 23.10 with nvidia proprietary driver (NVIDIA GeForce RTX 3050).
Xorg!

**Reply by brunoherbelin** [reply]
with nvidia, H264, H265 are supported with HW decoder nvh264dec and nvh265dec : most efficient IMHO.

**Comment by brunoherbelin** [comment]
CPU however should not be high, and from what I see in your screenshot, only one core is 60% active, two at 20%.. (thus a total of 4% CPU usage). Or do i wrongly read the top?

What I can confirm though is that, on Ubuntu with Nvidia, GPU fans are activated when vimix runs, even when not doing much.
To ensure nvidia decoders and  GPU is active on my computer (which has combined AMD-NVidia graphics card), I set NVIDIA settings PRIME profile to Performance mode. With the NVTOP app, I can see that nvidia drivers are demanded and fan is strong for not much. I once tried GreenWithEnvy app (Flathub) : it is good to observe how much nvidia GPU decoder is used with vimix, but i don't know if it is effective in customizing performance.

**Reply by lupin3rd** [reply]
Hi @brunoherbelin 
I did numerous tests as you suggested...
I thought that the lag and freeze problems were due to vimix, but that's not the case!
There is a bug in the linux kernel of ubuntu 23.10 with intel driver (only when you use secondary monitor):
https://gitlab.freedesktop.org/drm/intel/-/issues/8685
... and i use secondary monitor only with vimix :-)

**Comment by brunoherbelin** [comment]
So, after some investigation and performance testing following your comment, I could indeed find the cause for the *ONE* CPU busy when vimix does nothing !! This is fixed in Beta 7238eccfd25bfc6830b54d996240b1d78cccf8a8
i.e. when idle, vimix does not cause a busy CPU anymore.

For GPU with nvidia, there is a deeper problem with vsync and drivers in Ubuntu : I can't seem to have an actual v-sync opengl test program that keeps 60Hz... Doesn't seems to cause too much GPU activity though... I don't know.

**Reply by lupin3rd** [reply]
I tested now last beta.
I confirm that now CPU is not busy!

**Reply by lupin3rd** [reply]
I used vimix for a few hours today, I think the new beta has introduced a bug but I can't replicate it...
twice it happened to me that vimix crashed (freeze) with the CPU at 100%, I had to kill the process.
It happened both times when I went to the file menu with the mouse
I try to replcate it .... but don't find a secure way... but seem that the problem is when i open menù

**Reply by brunoherbelin** [reply]
humm... the fix for high CPU didn't change anything deep, it was just a forgotten object that was updating without need.. So I am not so much convinced this is the cause of your crash.

But for sure, there were several changes recently, so new bugs could have been introduced. Thanks for pulling latest commits (did some optimization) and keeping me in the loop if you encounter this bug again (or any other). Thanks for your understanding !

**Comment by lupin3rd** [comment]
Hi @brunoherbelin i tested a lot last beta... very light and stable... no more problem with CPU or freeze. Thank you.

**Reply by brunoherbelin** [reply]
Good! Thanks! We're getting there !!
